functions/index.js

```diff
@@ -1,156 +1,389 @@
+          headers: {
+            'Access-Control-Allow-Origin': '*'
+          }
+        });
+      }
+    })
+    .catch((err) => {
+      callback(null, {
+        statusCode: 400,
+        body: JSON.stringify({
+          message: `Error: ${err}`
+        })
+      })
+    });
   }
-}
 
-async function modifyGame(event){
-  let updateExpression = [];
-  const modifyParameters = [ 'description', 'genre', 'yearReleased', 'developer', 'console' ];
-  let expressionAttributeNames={};
-  let expressionAttributeValues = {};
-  
-  //Generate dynammic update expression based on allowed parameters
-  for (let parameter in event.queryStringParameters) {
-    if (modifyParameters.includes(parameter)) {
-      updateExpression.push(`#${parameter} = :${parameter}`);
-      expressionAttributeNames['#'+parameter] = parameter ;
-      expressionAttributeValues[':'+parameter]=event.queryStringParameters[parameter];  
+  async function deleteGame(event) {
+    await returnDeleteData(event)
+    .then((data) => {
+      if (data.hasOwnProperty('code') && data.hasOwnProperty('message') && data.hasOwnProperty('statusCode')) {
+        switch (data.code) {
+          default:
+            callback(null, {
+              statusCode: `${data.statusCode}`,
+              body: `Message: ${data.message}\nStatus Code: ${data.statusCode}`,
+              headers: {
+                'Access-Control-Allow-Origin': '*'
+              }
+            });
+            break;
+        }
+      } else if (data) {
+        //DynamoDB returns an empty JSON object when attempting to delete an item that isn't in the table
+        switch (Object.keys(data).length) {
+          case 0:
+            callback(null, {
+              statusCode: 404,
+              body: `Message: ${JSON.parse(event.body).gameName} not found\nStatus Code: 404`,
+              headers: {
+                'Access-Control-Allow-Origin': '*'
+              }
+            });
+            break;
+          default:
+            callback(null, {
+              statusCode: 200,
+              body: JSON.stringify(data.Attributes),
+              headers: {
+                'Access-Control-Allow-Origin': '*'
+              }
+            });
+        }
+      } else {
+        callback(null, {
+          statusCode: 400,
+          body: "There was an error.",
+          headers: {
+            'Access-Control-Allow-Origin': '*'
+          }
+        })
+      }
+    })
+    .catch((err) => {
+      callback(null, {
+        statusCode: 400,
+        body: JSON.stringify({
+          message: `Error: ${err}`
+        })
+      })
+    });
+  }
+
+  //DynamoDB Data Access Functions
+  async function returnGetData(event){
+    let params = {
+      TableName: table,
+      Key: {
+        gameName: event.queryStringParameters["gameName"]
+      }
+    };
+    
+    try {
+      let data = await docClient.get(params).promise();
+      return data;
+    } catch (err) {
+      return err;
```

Comment:
How would the calling function know an error occurred here? Finding a more consistent way to return errors (like using `throw` will simplify logic for the callers.